### PR TITLE
Document new build bundling options

### LINKED ISSUE #2215: Document the new bundle property options in polymer.json
Per the pending polymer-project-config 3.4.0 release and subsequent polymer-cli (version # TBD, probably 1.2.1), the `bundle` property of `build` specifications in `polymer.json` file newly supports a value that is an options object for the `polymer-bundler` instead of just using a default bundler configuration (`true`) or no bundler at all (`false`). 

See https://github.com/Polymer/polymer-project-config/pull/37 for details in the description of the new options.

### PR DESCRIPTION
**Pull request by katejeffreys**
Fixes #2215

### CODE REVIEW

**Comment by katejeffreys** [comment]
@usergenic Can has sanity check, when you're ready? :)

File: app/2.0/docs/tools/polymer-json.md [hunk 1]
```diff
@@ -183,6 +190,28 @@ And here is a configuration to generate two optimized builds: One bundled and on
...
+Here is a configuration to generate a bundled, minified application build with the following bundling options:
+
+* Specified paths are excluded from inlining
+* Comments are stripped
+* External CSS is not inlined
+* Identity source maps for inline scripts are created
+
+```json
+"build": [{
+  name: "bundled-custom",
```
Comment (usergenic) [inline app/2.0/docs/tools/polymer-json.md]: key "name" should be quoted to be valid json.

File: app/2.0/toolbox/build-for-production.md [hunk 2]
```diff
@@ -170,6 +170,38 @@ To build this configuration:
...
+This configuration generates a bundled, minified application build with the following bundling options:
+
+* Specified paths are excluded from inlining
+* Comments are stripped
+* External CSS is not inlined
+* Identity source maps for inline scripts are created
+
+```json
+"build": [{
+  name: "bundled-custom",
```
Comment (usergenic) [inline app/2.0/toolbox/build-for-production.md]: key "name" should be quoted.

File: app/2.0/docs/tools/polymer-json.md [hunk 3]
```diff
@@ -146,7 +146,14 @@ required.
...
 * `insertPrefetchLinks`: If `true`, insert prefetch link elements into your fragments so that all
 dependencies are prefetched immediately.
-* `bundle`: If `true`, bundle your application.
+* `bundle`: May be boolean, or an object bag of settings matching specified [`polymer-bundler`](https://github.com/Polymer/polymer-bundler) constructor options.
+  * If `true`, bundle your application.
+  * If an object is provided with further configuration, the options are:
+    * `excludes`: Optional, `String[]`. Excludes the listed paths of files or folders from inlining.
+    * `inlineCss`: Optional, `Boolean`. Specifies whether to inline external CSS file contents into `<style>` tags.
+    * `inlineScripts`: Optional, `Boolean`. Specifies whether to inline external JavaScript file contents into `<script>` tags.
+    * `rewriteUrlsInTemplates`: Optional, `Boolean`. Specifies whether to rewrite element attributes inside of templates when inlining HTML.
```
Comment (usergenic) [inline app/2.0/docs/tools/polymer-json.md]: As an aside, this feature was how Polymer 1.x did things and some people with Polymer 1.x projects and elements may need this.  To the extent you think that provides value, maybe a small mention that this emulates the Polymer 1.x behavior.

**Pull request merged**